### PR TITLE
RM-72354 Manage connector log files

### PR DESCRIPTION
**Pull request by tomerickson-es**
Updating terraform code to load the new log management script onto the connector, and create a cron job that runs the script weekly.

References RM-72354

### CODE REVIEW

File: modules/aws/awc/user-data.sh.tmpl [hunk 1]
```diff
@@ -73,9 +73,26 @@ then
...
+EOF
+chmod +x /etc/cron.weekly/connector_log_manager
+
 log "--> Executing provisioning script..."
 . ${provisioning_script}
 
+# EditShare-specific; done after the aws provisioning script because that installs python3
+#log "--> Installing dryable module for log manager script..."
+#/usr/bin/pip3 install dryable
+
```
Comment (danstreeter-es) [inline modules/aws/awc/user-data.sh.tmpl]: This comment can now be removed as is performed in the provisioning script on [Ln83](https://github.com/EditShareCo/cloud_deployment_scripts_hp_inc/pull/5/files#diff-78af4455cc58e7941605aef17b0a4f71ecc996bca6306026df15c6a8f2515a2aR83).

File: modules/aws/awc/vars.tf [hunk 2]
```diff
@@ -210,3 +210,9 @@ variable "win_std_instance_count" {
   description = "Number of Windows Standard Workstations"
   default     = 0
 }
+
+variable "log_manager_script" {
+  description = "Number of Windows Standard Workstations"
```
Comment (danstreeter-es) [inline modules/aws/awc/vars.tf]: This description is not correct.

File: modules/aws/awc/user-data.sh.tmpl [hunk 3]
```diff
@@ -73,9 +73,26 @@ then
...
     chmod +x ${provisioning_script}
     dos2unix ${provisioning_script}
+
+    # EditShare-specific
+    log "--> Downloading connector_log_manager.py from the bucket..."
+    aws s3 cp s3://${bucket_name}/${log_manager_script} .
+    chmod +x ${log_manager_script}
 fi
 
+# NOTE we may want to add this to crontab so we can more precisely control when it executes
```
Comment (danstreeter-es) [inline modules/aws/awc/user-data.sh.tmpl:83]: This comment is incorrect - we _are_ doing this now aren't we?

File: modules/aws/awc/awc-provisioning.sh.tmpl [hunk 4]
```diff
@@ -77,6 +77,70 @@ install_prereqs() {
...
+    fi
+
+    log "Enabling socat->dc connectivity so restarts if rebooted"
+    systemctl enable socat-dc
+    if [ $? -ne 0 ]
+    then
+        log "--> Exiting provisioning script due to an error."
+        exit 1
+    fi
+
```
Comment (danstreeter-es) [inline modules/aws/awc/awc-provisioning.sh.tmpl]: All of this is related to the `socat` '_backdoor_' we add for remote access.

Is it expected that this change is introduced here with the work for RM-72354 and the log management work?

File: modules/aws/awc/awc-provisioning.sh.tmpl [hunk 5]
```diff
@@ -431,6 +495,10 @@ cd $PROVISIONING_DIR
 
 install_prereqs
 
+install_socat
+
+install_socat_systemd_service
+
```
Comment (danstreeter-es) [inline modules/aws/awc/awc-provisioning.sh.tmpl]: Same comment re `socat` work introduction.
> Is it expected that we introduce this feature alongside this work?

File: modules/aws/awc/main.tf [hunk 6]
```diff
@@ -10,6 +10,7 @@ locals {
 
   provisioning_script = "awc-provisioning.sh"
   awm_script          = "get-connector-token.py"
+  log_manager_script  = "connector_log_manager.py"
```
Comment (danstreeter-es) [inline modules/aws/awc/main.tf:13]: The usage of `var.local_manager_script` and `local.local_manager_script` is mixed in the code.

A `local` is used within the `aws_s3_object.log-manager-script` resource on line [78](https://github.com/EditShareCo/cloud_deployment_scripts_hp_inc/pull/5/files#diff-51d281d110c60958957a4a5e352dd6e2aa0aa02de13680b3df9564a0dfbe6852R75) and the `local` used again in the resources list of it on line [172](https://github.com/EditShareCo/cloud_deployment_scripts_hp_inc/pull/5/files#diff-51d281d110c60958957a4a5e352dd6e2aa0aa02de13680b3df9564a0dfbe6852R172)
but the `variable` version of the same string is used in the `aws_s3_object.awc-provisioning-script` resource on line [121](https://github.com/EditShareCo/cloud_deployment_scripts_hp_inc/pull/5/files#diff-51d281d110c60958957a4a5e352dd6e2aa0aa02de13680b3df9564a0dfbe6852R121)

Is there a reason/purpose I have missed why we use a mixed economy and/or don't pass the module `var` through to the `local` so its used everywhere?

File: modules/aws/awc/main.tf [hunk 7]
```diff
@@ -70,6 +71,13 @@ resource "aws_s3_object" "tls-cert" {
   source = var.tls_cert
 }
 
+# EditShare-specific
+resource "aws_s3_object" "log-manager-script" {
+  bucket = var.bucket_name
+  key    = local.log_manager_script
+  source = "${path.module}/../../../../editshare_deployments/modules/connector/scripts/${local.log_manager_script}"
```
Comment (danstreeter-es) [inline modules/aws/awc/main.tf:78]: _Just a cross reference mention here to comment about mixed `local`/`var` usage on line [13](https://github.com/EditShareCo/cloud_deployment_scripts_hp_inc/pull/5/files#diff-51d281d110c60958957a4a5e352dd6e2aa0aa02de13680b3df9564a0dfbe6852R13)_

([Comment Link](https://github.com/EditShareCo/cloud_deployment_scripts_hp_inc/pull/5#discussion_r1445875484))

File: modules/aws/awc/main.tf [hunk 8]
```diff
@@ -110,6 +118,7 @@ data "template_file" "user-data" {
   vars = {
     bucket_name         = var.bucket_name,
     provisioning_script = local.provisioning_script,
+    log_manager_script  = var.log_manager_script,
```
Comment (danstreeter-es) [inline modules/aws/awc/main.tf]: _Just a cross reference mention here to comment about mixed `local`/`var` usage on line [13](https://github.com/EditShareCo/cloud_deployment_scripts_hp_inc/pull/5/files#diff-51d281d110c60958957a4a5e352dd6e2aa0aa02de13680b3df9564a0dfbe6852R13)_

([Comment Link](https://github.com/EditShareCo/cloud_deployment_scripts_hp_inc/pull/5#discussion_r1445875484))

File: modules/aws/awc/main.tf [hunk 9]
```diff
@@ -160,6 +169,7 @@ data "aws_iam_policy_document" "awc-policy-doc" {
     resources = [
       "arn:aws:s3:::${var.bucket_name}/${local.provisioning_script}",
       "arn:aws:s3:::${var.bucket_name}/${local.awm_script}",
+      "arn:aws:s3:::${var.bucket_name}/${local.log_manager_script}",
```
Comment (danstreeter-es) [inline modules/aws/awc/main.tf:172]: _Just a cross reference mention here to comment about mixed `local`/`var` usage on line [13](https://github.com/EditShareCo/cloud_deployment_scripts_hp_inc/pull/5/files#diff-51d281d110c60958957a4a5e352dd6e2aa0aa02de13680b3df9564a0dfbe6852R13)_

([Comment Link](https://github.com/EditShareCo/cloud_deployment_scripts_hp_inc/pull/5#discussion_r1445875484))

**Review by danstreeter-es — COMMENTED** [review]
A few comments here, sorry!

Feel like I'm being a bit picky around including 'non related work' in this commit - but may pay us back in the future when we try and figure out where we 'code to do X' and cant find it as we slid it into 'Commit for Y' work.

Also feel a bit rough picking up on the local/var stuff - Im half thinking its for a reason I've glossed over/forgotten and don't know about, so apologies if its a 'because you cant do it like that' answer and my recent distance from Terraform has caught up with me.

File: modules/aws/awc/main.tf (hunk 6 above)
Comment (tomerickson-es) [reply modules/aws/awc/main.tf:13]: changed to use var.log_manager_script throughout

File: modules/aws/awc/user-data.sh.tmpl (hunk 1 above)
Comment (tomerickson-es) [reply modules/aws/awc/user-data.sh.tmpl]: block removed

File: modules/aws/awc/vars.tf (hunk 2 above)
Comment (tomerickson-es) [reply modules/aws/awc/vars.tf]: description fixed

File: modules/aws/awc/user-data.sh.tmpl (hunk 3 above)
Comment (tomerickson-es) [reply modules/aws/awc/user-data.sh.tmpl:83]: The comment means doing the equivalent of `crontab -e` which edits the core cron file, and in there you can set specific start day and time.  We currently add a file in the cron.weekly directory which i think means it runs weekly from whenever it was first run (presumably when the job was first created).

File: modules/aws/awc/awc-provisioning.sh.tmpl (hunk 4 above)
Comment (tomerickson-es) [reply modules/aws/awc/awc-provisioning.sh.tmpl]: I don't know why this picked up the socat-related changes, and am open to suggestions on how to fix this.  If we don't have ideas, then i'm inclined to just check it in anyway knowing that the socat code is already in the master branch

Comment (danstreeter-es) [reply modules/aws/awc/awc-provisioning.sh.tmpl]: @tomerickson-es - As am I... I can see from a fresh checkout of this repos master branch that the code that is above is already in there... so as to why its coming out as a diff... 🤷 ...

File: modules/aws/awc/user-data.sh.tmpl (hunk 3 above)
Comment (danstreeter-es) [reply modules/aws/awc/user-data.sh.tmpl:83]: OK cool, Thought it was just a general mention to adding it to the cron, which we're doing. 👍

File: modules/aws/awc/main.tf (hunk 7 above)
Comment (tomerickson-es) [reply modules/aws/awc/main.tf:78]: general note that the use of var.log_manager_script has been removed, it was inconsistent with the way the rest of the script names were managed by the original teradici code, and caused more problems/changes than it was worth.  there should now only be references to local.log_manager_script, or to the underlying file name which is connector_log_manager.py

Comment (danstreeter-es) [reply modules/aws/awc/main.tf:78]: 👍 Seems sensible =)

**Review by danstreeter-es — APPROVED** [review]
Looking good. Merge and ship.

**Pull request merged**